utils/hare-shutdown

```diff
@@ -17,217 +17,78 @@
-    size = len(process_list)
-
-    if not size:
-        return
-    q: Queue = Queue(maxsize=size)
-    thread_count = min(thread_count, size)
-
-    worker_pool: List[Worker] = [Worker(q) for i in range(thread_count)]
-
-    for w in worker_pool:
-        w.start()
-
-    for proc in process_list:
-        q.put(StopProcess(process=proc))
-    for i in range(thread_count):
-        q.put(QuitMessage())
-
-    for worker in worker_pool:
-        worker.join()
-
-
-def main() -> None:
-    _setup_logging()
-    if not consul_is_active_at('localhost'):
-        if pcs_consul_is_active_at('localhost'):
+def check_cluster():
+    if not utils.consul_is_active_at('localhost'):
+        if utils.pcs_consul_is_active_at('localhost'):
             exit("Cluster is run by Pacemaker, try `pcs cluster stop --all'")
         else:
             exit('Cluster is not running')
 
+
+@repeat_if_fails()
+def _get_processes_and_node():
     cns = Consul()
+    processes = utils.processes_by_consul_svc_name(cns)
+    leader_node = utils.get_kv(cns, 'leader')
+    return (processes, leader_node)
 
-    @repeat_if_fails()
-    def _get_processes_and_node():
-        processes = processes_by_consul_svc_name(cns)
-        leader_node = get_kv(cns, 'leader')
-        return (processes, leader_node)
 
+def shutdown_cluster():
+    check_cluster()
     (processes, leader_node) = _get_processes_and_node()
-    for svc in shutdown_sequence:
+    for svc in utils.shutdown_sequence:
         # Some processes may be not started as per configuration
         if svc in processes:
-            _stop_parallel([proc for proc in processes[svc]])
+            utils.stop_parallel([proc for proc in processes[svc]])
 
-    if leader_node and not is_fake_leader_name(leader_node):
+    if leader_node and not utils.is_fake_leader_name(leader_node):
         logging.info(f'Shutting down RC Leader at {leader_node}... ')
         pkill: str = 'sudo pkill --exact -KILL consul &> /dev/null'
-        exec(ssh_prefix(leader_node) + pkill)
+        utils.exec_custom(utils.ssh_prefix(leader_node) + pkill)
+
+
+def local_node_shutdown():
+    check_cluster()
+    cns = Consul()
+    node = gethostname()
+    processes = utils.processes_node(cns, node)
+
+    for svc in utils.shutdown_sequence:
+        if svc in processes:
+            utils.stop_parallel([proc for proc in processes[svc]])
+
+
+def main() -> None:
+    try:
+        opts, args = getopt.getopt(sys.argv[1:],
+                                   "v",
+                                   ["verbose", "node"])
+    except getopt.GetoptError as err:
+        sys.exit(err)
+
+    shut_cluster = True
+    for (opt, value) in opts:
+        if opt in ['-v', '--verbose']:
+            logging.basicConfig(level=logging.DEBUG, format='%(message)s')
+        else:
+            logging.basicConfig(level=logging.INFO, format='%(message)s')
+        if opt in ['--node']:
+            shut_cluster = False
```

Comment:
no need of `shut_cluster` bool flag, invoke `shutdown_cluster()` in else part of `if opt in ['--node']` or return after `local_node_shutdown()`

Reply:
Done.